lib/ftp/listener.ex

```diff
@@ -0,0 +1,22 @@
+defmodule Ftp.Listener do
+    @moduledoc """
+    Module for testing event dispatcher.
+    TODO --> delete when testing is complete.
+    """
+  use GenServer
+  require Logger
+
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, [], name: __MODULE__)
+  end
+
+  def init(_args) do
+    for event <- Application.get_env(:ftp, :events), do: Ftp.EventDispatcher.register(event)
```

Comment:
INFORMATIONAL ONLY: You are not required to make a change this works great, and speedy at our levels.

The `for` collects output, which is great if you need it. It also has some other filters and things that are super awesome. Unfortunately, it isn't the hammer to use when you don't care about the return or crazy filtering. I ran a benchmark to double check my suspicions.

# 10,000 items

![screen shot 2018-07-19 at 1 03 44 pm](https://user-images.githubusercontent.com/4253/42961366-39d23608-8b54-11e8-9a02-f9263e8784a7.png)

# 10 items

![screen shot 2018-07-19 at 1 06 20 pm](https://user-images.githubusercontent.com/4253/42961557-adeb72b6-8b54-11e8-9fd6-d66081c0903e.png)

I did it with 7 items too since that is what we have in this list and the results are the same. The results are not as dramatic on speed

![screen shot 2018-07-19 at 1 12 52 pm](https://user-images.githubusercontent.com/4253/42961804-83130080-8b55-11e8-8ee6-6e39e45a7279.png)

Fortunately, we don't have the memory hit or much of a speed hit since we are using things that work with atoms. This can get quite expensive so watch out for this.

Reply:
That's really cool.